src/main/java/com/android/volley/AsyncCache.java

```diff
@@ -0,0 +1,82 @@
+package com.android.volley;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicReference;
+
+public abstract class AsyncCache implements Cache {
+
+    public interface OnGetCompleteCallback {
+        void onGetComplete(Entry entry);
+    }
+
+    public abstract void get(String key, OnGetCompleteCallback callback);
+
+    // TODO: Handle callback never being invoked? Match behavior when sync cache throws
+    // RuntimeException.
+
+    @Override
```

Comment:
Could you also annotate the method as `@Nullable`? It acts a documentation to tell the caller that the return value could be null.

Reply:
fixed